ethereum/extethclient/eth_wallet_client.go

```diff
@@ -51,12 +53,26 @@ type EthClient interface {
 	Lock()   // Lock the wallet so only one transaction runs at at time
 	Unlock() // Unlock the wallet after a transaction is complete
 
-	// transfers ETH to the given address
-	// does not need locking, as it locks internally
-	Transfer(ctx context.Context, to ethcommon.Address, amount *coins.WeiAmount) (ethcommon.Hash, error)
-
-	// attempts to cancel a transaction with the given nonce by sending a zero-value tx to ourselves
-	CancelTxWithNonce(ctx context.Context, nonce uint64, gasPrice *big.Int) (ethcommon.Hash, error)
+	// Transfer transfers ETH to the given address, nonce Lock()/Unlock()
+	// handling is done internally. The gasLimit field when the destination
+	// address is not a contract.
```

Comment:
I don't like having to maintain comments on both the interface and the matching functions in the same file. If you agree, I'll remove this interface in a follow-up PR, since it is just the full public API of the interface's only implementation.